src/parseTools.js

```diff
@@ -1566,11 +1566,19 @@ var POSIX_PAGE_SIZE = 16384;
 // function, to which we pass the parsed-out name, arguments, and body of the
 // function. Returns the output of that function.
 function modifyFunction(text, func) {
-  var match = text.match(/\s*function\s+([^(]*)?\s*\(([^)]*)\)/);
-  assert(match, 'could not match function ' + text + '.');
-  var name = match[1];
-  var args = match[2];
-  var rest = text.substr(match[0].length);
+  var match = text.match(/^\s*function\s+([^(]*)?\s*\(([^)]*)\)/);
+  var name, args, rest;
+  if (match) {
+    name = match[1];
+    args = match[2];
+    rest = text.substr(match[0].length);
+  } else {
+    match = text.match(/^\s*function\(([^)]*)\)/);
+    assert(match, 'could not match function ' + text + '.');
+    name = '';
+    args = match[1];
+    rest = text.substr(match[0].length);
+  }
```

Comment:
Can you explain this part of the change?  Maybe add a comment?

Reply:
This generalizes the parsing of functions as text - specifically to handle a function without a name. (There's probably a fancier regex way to do this, but this seems less ambiguous.) I'll add a comment.